packages/odyssey-react-mui/src/RadioGroup.tsx

```diff
@@ -73,31 +81,64 @@ const RadioGroup = ({
   HintLinkComponent,
   id: idOverride,
   isDisabled,
+  isReadOnly = false,
   label,
   name: nameOverride,
   onChange: onChangeProp,
   testId,
   translate,
   value,
 }: RadioGroupProps) => {
+  const [selectedValue, setSelectedValue] = useState(value ?? defaultValue);
```

Comment:
Might be missing something but not sure why we need to track the value internally for this component

Reply:
I refactored this a bit in the commit I just pushed, but the purpose of this is for the uncontrolled case since its not managed by the MUI component itself. Without it, the read-only logic won't work because it will just update when a new selection is made.